src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,38 @@
+package core.basesyntax;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAGIC_NUMBER = 10;
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        ColorSupplier supplier = new ColorSupplier();
+        Figure f1 = new Circle(getRandomProperty(), supplier.getRandomColor());
+        Figure f2 = new Square(getRandomProperty(), getRandomProperty(),
+                supplier.getRandomColor());
+        Figure f3 = new RightTriangle(getRandomProperty(), getRandomProperty(),
+                getRandomProperty(), supplier.getRandomColor());
```

Comment:
The `RightTriangle` constructor takes only 3 parameters, but you're passing 4. Remove the extra parameter to match the constructor signature.